src/pages/registrationPage/_tests_/RegistrationPage.spec.tsx

```diff
@@ -0,0 +1,53 @@
+import { fireEvent, render, screen, waitFor, } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import { MemoryRouter } from 'react-router-dom';
+import RegistrationPage from '../registrationPage';
+
+test('renders registration page', () => {
+  render(
+    <MemoryRouter>
+      <RegistrationPage />
+    </MemoryRouter>,
+  );
+
+  expect(screen.getByText(/Register your account/i)).toBeInTheDocument();
+  expect(screen.getByPlaceholderText(/username/i)).toBeInTheDocument();
+  expect(screen.getByPlaceholderText(/user@gmail.com/i)).toBeInTheDocument();
+  expect(screen.getByText(/Confirm Password/i)).toBeInTheDocument();
+  expect(screen.getByText(/Show password/i)).toBeInTheDocument();
+  expect(screen.getByText(/Register like admin/i)).toBeInTheDocument();
+  expect(screen.getByText(/Have account?/i)).toBeInTheDocument();
+});
+
+describe('RegistrationPage Component', () => {
+    test('toggles password visibility back and forth', () => {
+        render(
```

Comment:
If the logic for rendering the `RegistrationPage` is common across tests, you might consider extracting it into a `beforeEach` block to reduce duplication.